openzeppelin/security/reentrancy_guard.cairo

```diff
@@ -0,0 +1,35 @@
+# SPDX-License-Identifier: MIT
+# OpenZeppelin Cairo Contracts v0.1.0 (security/reentrancyguard.cairo)
+
+%lang starknet
+
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+
+from openzeppelin.utils.constants import TRUE, FALSE
+
+@storage_var
+func _entered() -> (res: felt):
+end
```

Comment:
Does it clash in memory if it has the same name as other storage var in another module? I honestly only thought about prefixing storage that's meant to be imported by users, otherwise being private storage of the module. But if collisions are possible then we should prefix everything.

Reply:
Ahh I misunderstood the reason for prefixing storage. I'm incorrect here.

Reply:
_But does it clash_ if I have two working `_variable_name` storage functions being used "privately" in different modules?

Reply:
@martriay thank you for pushing on that question! It appears that the "private" storage variables _will_ clash if they share the same name. I created a gist to duplicate my experiment [here](https://gist.github.com/andrew-fleming/6d99d68a55be6228b0b5c1ca4c58dc90).

Reply:
TL;DR:

The `balance` storage variable of `a` and `b` is seen as the same variable in `c`; so when adding 99 to `a` and `100` to `b`, `c`'s output for `get_balance` on both variables is `199`. 

I also tried removing the `amount` parameters and hardcoded that `a` increases by `99` and `b` increases by `100`. The `balance` variable still gets overwritten by the last `increase_balance` method called.